lib/spack/spack/environment.py

```diff
@@ -557,7 +557,11 @@ def regenerate(self, all_specs, roots):
             if not os.path.isabs(root):
                 root = os.path.normpath(os.path.join(self.base, self.root))
             fs.mkdirp(root)
-            with fs.replace_directory_transaction(root):
+
+            # The tempdir for the directory transaction must be in the same 
+            # filesystem mount as the view, so that view creation fails to 
+            # symlink only when the finished view would fail as well
```

Comment:
A couple requests on this comment:

* Update it to indicate precisely what it is addressing. In this case the fact that `replace_directory_transaction` is being called with two arguments. Something like `# Provide dirname(root) as the staging area for the replace_directory_transaction because...
* The second part of this comment `... so that view creation fails to symlink only when the finished view would fail as well` is a bit vague. I think it would be sufficient and clearer to mention the example prompting this change in the first place:

> Provide dirname(root) as the staging area for the replace_directory_transaction because it must be on the same filesystem mount as the view itself: it may not be possible to symlink between filesystem mounts, so if the default staging area of tmpdir is used the operation could fail.

While I think `so that view creation fails to symlink only when the finished view would fail as well` captures more details, it takes more effort to unpack.

Reply:
@scheibelp I've addressed this review now. Sorry about the delay.